DNN Platform/Library/Application/ApplicationStatusInfo.cs

```diff
@@ -38,6 +38,7 @@ public ApplicationStatusInfo(IApplicationInfo applicationInfo)
         }
 
         /// <inheritdoc />
+        [CLSCompliant(false)]
```

Comment:
I accepted all other suggestions, this one I am a bit confused, I get a warning here that Status is not CLS compliant and I am not sure:
1. Why that is not CSL compliant
2. How making other assemblies CLS compliant would solve that
3. Where I would need to put the [asssembly: CLSCompliant(true)] suggestion above to even try it out...

Reply:
It's not CLS compliant because it uses a type, `UpgradeStatus`, which is not marked as CLS compliant.  That type isn't CLS compliant because it's defined in a project, `DotNetNuke.Abstractions`, that's not marked as CLS compliant.  There's a warning because `ApplicationStatusInfo` _is_ marked as being CLS Compliant, because its project is marked as being compliant, via an assembly attribute in [`AssemblyInfo.cs`](https://github.com/dnnsoftware/Dnn.Platform/blob/cc61549005ab5ea1c66fa91670931642a62893e4/DNN%20Platform/Library/Properties/AssemblyInfo.cs#L17).

You would need to add a `Properties\AssemblyInfo.cs` file to `DotNetNuke.Abstractions` and place the assembly attribute in there (you can really place it anywhere, but that's the conventional place to put assembly attributes that aren't otherwise related to a specific type).

Reply:
Oh, this is awesome and it worked out perfect. Thanks for the info, we learn every day!

Reply:
Just for clarity, I implemented your suggestion and removed the false one and added 2 true ones for DependencyInjection and Abstractions